docs/protocol/extension_logging.md

```diff
@@ -0,0 +1,198 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Logging Extension
+
+This document describes Triton's logging extension. The logging extension enables
+the client to configure log settings during a Triton run. Triton reports "logging" 
+in the extensions field of its Server Metadata.
+
+## HTTP/REST
+
+In all JSON schemas shown in this document $number, $string, $boolean,
+$object and $array refer to the fundamental JSON types. #optional
+indicates an optional JSON field.
+
+Triton exposes the logging endpoint at the following URL. The client may use
+HTTP GET request to retrieve the current log settings. A HTTP POST request
+will modify the log settings, and the endpoint will return the updated log
+settings on success or an error in the case of failure. 
+
+```
+GET v2/logging
+
+POST v2/logging
+```
+
+### Log Setting Response JSON Object
+
+A successful log setting request is indicated by a 200 HTTP status
+code. The response object, identified as $log_setting_response, is
+returned in the HTTP body for every successful log setting request.
+
+```
+$log_setting_response =
+{
+  $log_setting, ...
+}
+
+$log_setting = $string : $string | $boolean | $uint32
+```
+
+Each $log_setting JSON describes a “name”/”value” pair, where the “name” is
+the $string representation of the log setting and the “value” is a $string, $bool, or $uint32 representation of the
+setting value. Currently, the following log settings are defined:
+
+- "log_file" : a $string parameter defining the file where the log outputs will be saved. If not specified,
+log outputs will stream to the console.
+
+- "log_info" : a $boolean parameter that controls whether the Triton server logs INFO level messages. 
+
+- "log_warnings" : a $boolean parameter that controls whether the Triton server logs WARNING level messages. 
+
+- "log_errors" : a $boolean parameter that controls whether the Triton server logs ERROR level messages. 
+
+- "log_verbose_level" : a $uint32 parameter that controls whether the Triton server outputs verbose messages
+of varying degrees. This value can be any integer >= 0. If "log_verbose_level" is 0, no verbose messages
+will be output by the Triton server. If "log_verbose_level" is 1, level 1 verbose messages will be output
+by the Triton server. If "log_verbose_level" is 2, the Triton server will output all verbose messages of 
+level <= 2, etc.
+
+- "log_format" : a $string parameter that controls the format of Triton server log messages. There are currently
+2 formats: "default" and "ISO8601".
+
+
+### Log Setting Response JSON Error Object
+
+A failed log setting request will be indicated by an HTTP error status
+(typically 400). The HTTP body will contain a
+$log_setting_error_response object.
+
+```
+$log_setting_error_response =
+{
+  "error": $string
+}
+```
+
+- “error” : The descriptive message for the error.
+
+#### Log Setting Request JSON Object
```

Comment:
I think this should be header3 on par with response sections, not header4
```suggestion
### Log Setting Request JSON Object
```

Reply:
You're correct. I used the trace extension file as a template for this one and looking back, several of the issues you've mentioned are present in other extension files. Would you recommend opening PRs for them or keeping them in mind for their next update?